File: backend/src/main/java/com/backend/domain/resume/repository/ResumeRepository.java
package com.backend.domain.resume.repository;

import com.backend.domain.resume.entity.Resume;
import com.backend.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ResumeRepository extends JpaRepository<Resume, Long> {
    Boolean existsByUserId(Long userId);
    Optional<Resume> findByUser(User user);
}

File: backend/src/main/java/com/backend/domain/resume/repository/ResumeRepository.kt
package com.backend.domain.resume.repository

import com.backend.domain.resume.entity.Resume
import com.backend.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface ResumeRepository : JpaRepository<Resume, Long> {
    fun existsByUserId(userId: Long): Boolean
    fun findByUser(user: User): Resume?
}
